### ISSUE TITLE
Variants - core dump 

### ISSUE THREAD

**Issue by cinaljess** [jts/nanopolish#197]
Hello Jared, 
  Thanks for the great tool - I hope I can get it working as it looks very promising!

I compiled nanopolish using GCC-5.2 and make. I received the error immediately after execution.

**Execution** 

nanopolish variants --consensus nanopolish_ACall/polished.0-4120.fa -w 0-4120 -r nanopolish_ACall/arturo_all_reads.fa -b nanopolish_ACall/NB01.consensus.sorted.bam -g nanopolish_ACall/drafts/NB01.consensus.fasta --min-candidate-depth 100 --models-fofn nanopolish_models.fofn

**Stderr**

#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
TODO: train model
TODO: filter data
nanopolish: src/common/nanopolish_fast5_map.cpp:84: void Fast5Map::load_from_fasta(std::__cxx11::string): Assertion `read_to_path_map.find(seq->name.s) == read_to_path_map.end()' failed.
Aborted (core dumped)

**Core dump**

registering model t.007-base
registering model t.007-ONT
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
TODO: train model
TODO: filter data
nanopolish: src/common/nanopolish_fast5_map.cpp:84: void Fast5Map::load_from_fasta(std::__cxx11::string): Assertion `read_to_path_map.find(seq->name.s) == read_to_path_map.end()' failed.

Program received signal SIGABRT, Aborted.
0x0000003bef832495 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x0000003bef832495 in raise () from /lib64/libc.so.6
#1  0x0000003bef833c75 in abort () from /lib64/libc.so.6
#2  0x0000003bef82b60e in __assert_fail_base () from /lib64/libc.so.6
#3  0x0000003bef82b6d0 in __assert_fail () from /lib64/libc.so.6
#4  0x000000000048db21 in Fast5Map::load_from_fasta (this=this@entry=0x7fffffffa1d0, fasta_filename=...) at src/common/nanopolish_fast5_map.cpp:84
#5  0x000000000048ef36 in Fast5Map::Fast5Map (this=0x7fffffffa1d0, fasta_filename=...) at src/common/nanopolish_fast5_map.cpp:38
#6  0x00000000004a1955 in AlignmentDB::AlignmentDB (this=0x7fffffffa120, reads_file=..., reference_file=..., sequence_bam=..., event_bam=..., calibrate_reads=<optimized out>) at src/alignment/nanopolish_alignment_db.cpp:35
#7  0x00000000004159ad in call_variants_for_region (contig=..., region_start=4120, region_end=-2) at src/nanopolish_call_variants.cpp:797
#8  0x00000000004172cf in call_variants_main (argc=<optimized out>, argv=<optimized out>) at src/nanopolish_call_variants.cpp:1012
#9  0x0000000000408860 in std::function<int (int, char**)>::operator()(int, char**) const (__args#1=0x7fffffffa710, __args#0=15, this=0x9abf50)
    at /broad/software/free/Linux/redhat_6_x86_64/pkgs/gcc_5.2.0/include/c++/5.2.0/functional:2271
#10 main (argc=16, argv=0x7fffffffa708) at src/main/nanopolish.cpp:59

**Prior commands**

bwa mem -x ont2d nanopolish_ACall/drafts/NB01.consensus.fasta nanopolish_ACall/arturo_all_reads.fa | samtools sort -o nanopolish_ACall/NB01.consensus.sorted.bam -T nanopolish_ACall/NB01.consensus.sorted.tmp -

**Files**
head -n 1 nanopolish_ACall/arturo_all_reads.fa
'>channel:1D_000:template GP1B9_B89_20170301_FNFAB30680_MN16958_sequencing_run_2nd_Attempt_AC_Samples_48030_ch245_read8000_strand arturo_all_source/fast5_all/GP1B9_B89_20170301_FNFAB30680_MN16958_sequencing_run_2nd_Attempt_AC_Samples_48030_ch245_read8000_strand.fast5'

**Comment by jts** [comment]
Hi,

Your fasta has duplicated entries, which prevents nanopolish from creating a read-to-fast5 map. Your reads are probably all named `channel:1D_000:template` but they should be named like `<unique-id>:1D_000:template`. What basecaller/extraction program did you use?

Jared

**Comment by cinaljess** [comment]
nanopolish extract --type template arturo_all_source/fast5_all/ > nanopolish_ACall/arturo_all_reads.fa

The fast5 files were obtained using Metrichor, skipping the cloud upload.

This was done on a directory that I created to hold all of the template fast5 reads created by copying over from the 1/, 2/, 3/, ... 22/ directories of fast5 reads.

You're right 1D_000:template is the first string in all 87,000 reads. I have replaced 'channel' with an incrementing integer and will re-try the pipeline.
